tests/unit_tests/sql/parse_tests.py

```diff
@@ -1625,3 +1642,608 @@ def test_as_cte(sql: str, engine: str, expected: str) -> None:
     Test that we can covert select to CTE.
     """
     assert SQLStatement(sql, engine).as_cte().format() == expected
+
+
+@pytest.mark.parametrize(
```

Comment:
These tests are all copied from `superset/tests/unit_tests/sql_parse_tests.py`, except a few that are not relevant.